Source/Serilog.Enrichers.Span/ActivityTraceStateEnricher.cs

```diff
@@ -0,0 +1,39 @@
+namespace Serilog.Enrichers.Span;
+
+using System;
+using System.Diagnostics;
+using System.Linq;
+using Serilog.Core;
+using Serilog.Events;
+
+/// <summary>
+/// A log event enricher which adds trace state from the current <see cref="Activity"/>.
+/// </summary>
+public class ActivityTraceStateEnricher : ILogEventEnricher
+{
+    /// <summary>
+    /// Enrich the log event.
+    /// </summary>
+    /// <param name="logEvent">The log event to enrich.</param>
+    /// <param name="propertyFactory">Factory for creating new properties to add to the event.</param>
+    public void Enrich(LogEvent logEvent, ILogEventPropertyFactory propertyFactory)
+    {
+#if NET6_0_OR_GREATER
+        ArgumentNullException.ThrowIfNull(logEvent);
+#else
+        if (logEvent is null)
+        {
+            throw new ArgumentNullException(nameof(logEvent));
+        }
+#endif
+
+        if (Activity.Current is not null && !string.IsNullOrWhiteSpace(Activity.Current.TraceStateString))
+        {
+            var state = Activity.Current.TraceStateString?
```

Comment:
Does the `TraceStateString` always conform this form? Since its just a string, presumably it could be set to anything.

Reply:
It'd be safer to just log the string rather than try to break it down but does that make sense?

Reply:
Yes according to [W3C spec](https://www.w3.org/TR/trace-context/#tracestate-header) trace state header does always conform to this format. The reasons for Activity class not providing this parsing logic itself is mentioned here https://github.com/dotnet/runtime/issues/26463#issuecomment-397427815 basically due to a desire `to separate the 'HTTP-like stuff' from the Activity class`.

The definition for key and value indicate comma and equals character is not allowed and there's no mention of a mechanism to escape characters so don't need to deal with un-escaping characters. However it does mention whitespace before and after key or value is allowed and empty list members is allowed as well, so should improve above code by trimming key/value and filtering out empty list members.